### PR TITLE
Add documentation for "includeTriggerModifications"

### PR DESCRIPTION
**Pull request by KalebMatthews**

### CODE REVIEW

File: sections/interfaces.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default [
     type: "method",
     method: "then",
     example: ".then(onFulfilled, [onRejected])",
-    description: "Coerces the current query builder chain into a promise state, accepting the resolve and reject handlers as specified by the Promises/A+ spec. As stated in the spec, more than one call to the then method for the current query chain will resolve with the same value, in the order they were called; the query will not be executed multiple times.",
+    description: "Coerces the current query builder chain into a promise state, accepting the resolve and reject handlers as specified by the Promises/A+ spec. As stated in the spec, more than one call to the then method for the current query chain will resolve with the same value, in the order they were called; the query will not be executed multiple times. An additional option can be included if the table has triggers associated with it.",
```
Comment (kibertoad) [inline sections/interfaces.js:50]: I don't think this is the right place.
Documentation for `insert`, `delete` and `update` operations should be changed, this is not a change to `then` operation.

Comment (KalebMatthews) [reply sections/interfaces.js]: My bad. I found it. Didn't know exactly where everything was.

**Review by kibertoad — CHANGES_REQUESTED** [review]
Signature of all affected methods needs to be changed as well.
e. g. `insert — .insert(data, [returning])`

See other operations with parameters as an example, e. g.
`enum / enu — table.enu(col, values, [options])`

Also explanation for `includeTriggerModifications` parameter should be added, explaining that it affects the `returning` part of the query, and that it is only supported by MSSQL.

Shouldn't `returning` operation be changed to support this parameter as well?

**Comment by KalebMatthews** [comment]
@kibertoad Do you want the example section to be a runnable type or a code type? I am assuming that this means it generates an output statement or not.

**Comment by kibertoad** [comment]
@KalebMatthews Considering that output would be very MSSQL-specific, I'd say that code without output is better.

File: sections/builder.js [hunk 2]
```diff
@@ -1823,18 +1830,25 @@ export default [
         `
       },
       {
-        type: "runnable",
+        type: "text",
+        content: "For MSSQL, triggers on tables can interrupt returning a valid value from the standard insert statements. You can add the `includeTriggerModifications` option to get around this issue. This modifies the SQL so the proper values can be returned. This only modifies the statement if you are using MSSQL, a returning value is specified, and the `includeTriggerModifications` option is set."
```
Comment (kibertoad) [inline sections/builder.js:1834]: shouldn't it be `update statements` here?

File: sections/builder.js [hunk 3]
```diff
@@ -1846,10 +1860,18 @@ export default [
...
```
Comment (kibertoad) [inline sections/builder.js:1864]: shouldn't it be `delete statements`?

Comment (kibertoad) [reply sections/builder.js:1864]: btw, what does `returning` for deletion even do? is it supported at all?

Comment (KalebMatthews) [reply sections/builder.js]: From my testing, yes, it does work and it is covered in test scenarios already written.

**Comment by kibertoad** [comment]
@KalebMatthews This needs to be updated for `returning` operation, and then your titantic effort will be completed. Thank you so much for making it happen!

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thanks! Will merge after CI on the code PR is green.

**Pull request merged**